src/main/java/uk/gov/hmcts/ccd/domain/service/processor/GlobalSearchProcessorService.java

```diff
@@ -0,0 +1,160 @@
+package uk.gov.hmcts.ccd.domain.service.processor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.config.JacksonUtils;
+import uk.gov.hmcts.ccd.domain.model.common.CaseFieldPathUtils;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.globalsearch.OtherCaseReference;
+import uk.gov.hmcts.ccd.domain.model.globalsearch.SearchCriteria;
+import uk.gov.hmcts.ccd.domain.model.globalsearch.SearchParty;
+import uk.gov.hmcts.ccd.domain.model.globalsearch.SearchPartyValue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@Service
+public class GlobalSearchProcessorService {
+
+    public Map<String, JsonNode> populateGlobalSearchData(CaseTypeDefinition caseTypeDefinition,
+                                                          Map<String, JsonNode> data) {
+        List<uk.gov.hmcts.ccd.domain.model.definition.SearchCriteria> searchCriterias =
+            caseTypeDefinition.getSearchCriterias();
+
+        List<uk.gov.hmcts.ccd.domain.model.definition.SearchParty> searchParties =
+            caseTypeDefinition.getSearchParties();
+
+        SearchCriteria searchCriteria = populateSearchCriteria(data, searchCriterias);
+        List<SearchParty> searchPartyList = populateSearchParties(data, searchParties);
+
+        if (searchCriteria.isEmpty() && !searchPartyList.isEmpty()) {
+            searchCriteria = SearchCriteria.builder().searchParties(searchPartyList).build();
+        }
+
+        if (!searchCriteria.isEmpty() && data != null) {
+            data.put("SearchCriteria", JacksonUtils.convertValueJsonNode(searchCriteria));
+        }
+
+        return data;
```

Comment:
This method is behaving like a void, it takes in 'data' as a parameter and returns 'data'.  I think it should avoid mutating the input parameter and then returning it.  Would be a problem if the caller passes in an unmodifiable map, I suspect.

Reply:
Understood intention a bit clearer after discussion - addressed in https://github.com/hmcts/ccd-data-store-api/pull/1637/commits/6918ba08bf87e46b7f41d1840da9716dbb33af42